ios/Podfile

```diff
@@ -0,0 +1,37 @@
+# This was the iOS Deployment Target (project > ZulipMobile > Info in Xcode)
+# as of 2020-03.
+platform :ios, '10.3'
+
+require_relative '../node_modules/react-native-unimodules/cocoapods.rb'
+
+# See
+# https://github.com/facebook/react-native-website/blob/ded79d2cf4456d8b1a4f67c2cdc1391789e70617/docs/integration-with-existing-apps.md#configuring-cocoapods-dependencies
+# for the choice of pods here.
+target 'ZulipMobile' do
+  pod 'React', :path => '../node_modules/react-native', :subspecs => [
+    'Core',
+    'CxxBridge', # Include this for RN >= 0.47
+    'DevSupport', # Include this to enable In-App Devmenu if RN >= 0.43
+    'RCTText',
+    'RCTNetwork',
+    'RCTWebSocket', # Needed for debugging
+    'RCTAnimation', # Needed for FlatList and animations running on native UI thread
+    'RCTImage', # Included in react-native-unimodules' example Podfile
+    # Add any other subspecs you want to use in your project
+  ]
+  # Explicitly include Yoga if you are using RN >= 0.42.0
+  pod 'yoga', :path => '../node_modules/react-native/ReactCommon/yoga'
+
+  # Third party deps podspec link
+  pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
+  pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec'
+  pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'
+
+  use_unimodules!
```

Comment:
Mm, I thought I caught that, but obviously not.

Originally, I did everything at once — the CocoaPods work and adding unimodules — and made a single commit. Then, when I thought it would be good to split that into two commits, my strategy was to create a commit after the original commit, deleting the code that added unimodules. And then to reverse the commits.

But to accomplish that reversal, I knew that a simple interactive rebase with the two `pick` commands reversed wouldn't work. That's because the "pick" command applies a computed patch, it doesn't just effect the selected commit's snapshot. Usually applying a patch is what we want, but in this case, I had snapshot A and snapshot B, and I simply wanted to reorder them so B came before A.

I ended up doing an interactive rebase and breaking before the first commit, then doing `git checkout <secondCommit> -- .`, committing that, then `git checkout <firstCommit> -- .`, committing that, then `git rebase --edit-todo`, I dropped the two original commits, and finished the rebase. I'm not sure if this was exactly where this mistake was introduced, but it sure seemed like a lot of maneuvering just to reverse the two commits. Do you know of a better way?

Reply:
> Usually applying a patch is what we want, but in this case, I had snapshot A and snapshot B, and I simply wanted to reorder them so B came before A.

One way I might do this is:
```
git reset --soft @^^
# Now your HEAD includes neither commit, but your worktree and index are still at B.

git commit -c @{1}
# Now you have one commit, with tree/snapshot B.

git checkout @{2}^ .
# Now the worktree and index are at A.

git commit -c @{2}^
# Now you have both commits.
```

The `-c` are optional, but convenient for preserving the author dates and commit messages (but still give you an editor to edit the commit messages.)

The @^^ and @{2}^ and so on could be replaced with explicit commit IDs copied from `git log`, but make handy ways to refer to the commits without needing to copy-paste.

When the commits in question are in the middle of a longer series, I might use `git rebase -i` with just a `b` / `break` command inserted after the two commits, and then do the above steps there.

Reply:
Thanks, I've bookmarked this!